sharding/data_modeling.rst

```diff
@@ -24,12 +24,77 @@ If your situation resembles either of these cases then the next step is to decid
 
-The first step is identifying what constitutes a tenant in your app. Common instances include company, account, organization, or customer. The column name will be something like :code:`company_id` or :code:`customer_id`. Examine each of your queries and ask yourself: would it work if it had additional WHERE clauses to restrict all tables involved to rows with the same tenant id? Queries in the multi-tenant model are usually scoped to a tenant, for instance queries on sales or inventory would be scoped within a certain store.
+Here is an example of two tables and their shards. Each box is a shard whose color represents which physical node contains it. Green shards are stored together on one physical node, and blue on another.  Notice how a join query between Accounts and Campaigns would have all the necessary data together on one node when restricting both tables to the same account_id. Matching the relevant shards of two tables in this way across all nodes is called *table co-location*.
+
+.. figure:: ../images/mt-colocation.png
+   :alt: co-located tables in multi-tenant architecture
+
+
+To apply this design in your own schema the first step is identifying what constitutes a tenant in your application. Common instances include company, account, organization, or customer. The column name will be something like :code:`company_id` or :code:`customer_id`. Examine each of your queries and ask yourself: would it work if it had additional WHERE clauses to restrict all tables involved to rows with the same tenant id? Queries in the multi-tenant model are usually scoped to a tenant, for instance queries on sales or inventory would be scoped within a certain store.
 
 If you're migrating an existing database to the Citus multi-tenant architecture then some of your tables may lack a column for the application-specific tenant id. You will need to add one and fill it with the correct values. This will denormalize your tables slightly. For more details and a concrete example of backfilling the tenant id, see our guide to :ref:`Multi-Tenant Migration <transitioning_mt>`.
 
+Typical Multi-Tenant Scenario
+-----------------------------
+
+Most SaaS applications already have the notion of tenancy built into their data model. In the following, we will look at an example schema from the online advertising space. In this example, a web advertising platform has tenants that it refers to as accounts. Each account holds and tracks advertising clicks across various campaigns.
+
+.. code-block:: postgres
+
+  CREATE TABLE accounts (
+    id integer PRIMARY KEY,
```

Comment:
Two quick questions.

1. Why not create a primary key on this table?
2. integer -> bigint for identifier columns?